tempesta_fw/http_parser.c

```diff
@@ -489,7 +490,12 @@ parse_int_hex(unsigned char *data, size_t len, unsigned long *acc)
 			return CSTR_NEQ;
 		if (unlikely(*acc > (UINT_MAX - 16) / 16))
 			return CSTR_BADLEN;
+		if (unlikely(*d_count >= (sizeof(*acc) - 1))) {
```

Comment:
You save half a byte on every read operation. Condition above allows you to store only 3 bytes. I doubt that it was a desired behaviour.